pkg/erofs/erofs.go

```diff
@@ -690,87 +725,162 @@ func (i *Inode) blocks() uint64 {
 //
 // [ (metadata block) inode | optional fields | dirent M+2 | dirent M+3 | name M+2 | name M+3 | optional padding ]
 //
-// All directory entries are _strictly_ recorded in alphabetical order.
-func (i *Inode) IterDirents(cb func(name string, typ uint8, nid uint64) error) error {
+// refer: https://docs.kernel.org/filesystems/erofs.html#directories
+func (i *Inode) getDirentName(d *Dirent, block blockData, lastDirent bool) ([]byte, *Dirent, error) {
+	var (
+		next    *Dirent
```

Comment:
Here too we are returning `next`, which seems unintuitive from the function name. 2/3 callers don't need it. `direntAfter()` should be a simple operation (just increment pointer by DirentSize), so I think it is okay if that work is repeated in IterDirents.

Reply:
Makes sense. Will do.